Managers/PrimitiveMan.h

```diff
@@ -0,0 +1,546 @@
+#ifndef _RTEPRIMITIVEMAN_
+#define _RTEPRIMITIVEMAN_
+
+#include "Singleton.h"
+#include "Vector.h"
+
+#define g_PrimitiveMan PrimitiveMan::Instance()
+
+struct BITMAP;
+
+namespace RTE {
+
+	class Entity;
+
+	/// <summary>
+	/// Singleton manager responsible for all primitive drawing.
+	/// </summary>
+	class PrimitiveMan : public Singleton<PrimitiveMan> {
+
+	public:
+
+#pragma region Primitive Objects
+#pragma region Graphical Primitive
+		/// <summary>
+		/// Class used to schedule drawing of graphical primitives created from Lua.
+		/// All coordinates passed to GraphicalPrimitive objects are Scene coordinates.
+		/// </summary>
+		class GraphicalPrimitive {
```

Comment:
They are only relevant to PrimitiveMan and I think it's best to keep them here because you don't ever use them elsewhere.

Reply:
You could make the same argument about, say PerformanceMan only being used in FrameMan. They bloat the header a bunch with stuff that could be separated and make everything a lot nicer to read through (the implementations can still happen in PerformanceMan.cpp just fine if we want), and they make the region stuff look ugly with
```
#pragma region Primitive Objects
#pragma region Graphical Primitive
```

Reply:
It may look ugly but that was done specifically so you can collapse the whole block and not have the eye sore.
I guess I can throw a new header in System for them but that's just giving these more attention than they deserve.

Reply:
Oh yeah, I get why you did it but I'd rather the new header option. IMO that puts them more out of the way since they're just in some quiet little include you never need to look in or think about, vs them filling up 3/4ths of PrimitiveMan.
I don't really like internal classes in cpp generally, I think they look bad and I often misread them as being something belonging to the main class (especially things like MOSRotating::Gib::Create vs MOSRotating::Create), and I think we should try to avoid them where possible to make the code clearer.